target/linux/ramips/patches-6.1/800-dmaengine-mediatek-add-HSDMA-support-for-mt7621.patch

```diff
@@ -0,0 +1,44 @@
+From d94fc5ce1dc395747c3934ecffcdec0396583755 Mon Sep 17 00:00:00 2001
+From: Nick Hainke <vincent@systemli.org>
+Date: Fri, 26 May 2023 19:46:33 +0200
+Subject: [PATCH] dmaengine: mediatek: add HSDMA support for mt7621
+
+Commit 87dd67f496f7 ("staging: mt7621-dma: remove driver from tree")
+removed the mt7621-dma driver. Move the driver from staging to the
+folder "drivers/dma/mediatek" containing already other mediatek dma
+driver implementations and maintain it downstream in OpenWrt.
+
+This patch will not be sent to upstream linux. It is just a workaround.
+
+Signed-off-by: Nick Hainke <vincent@systemli.org>
+---
+ drivers/dma/mediatek/Kconfig  | 6 ++++++
+ drivers/dma/mediatek/Makefile | 1 +
+ 2 files changed, 7 insertions(+)
+
+diff --git a/drivers/dma/mediatek/Kconfig b/drivers/dma/mediatek/Kconfig
+index 7a46a5455957..1a24d73ce87f 100644
+--- a/drivers/dma/mediatek/Kconfig
++++ b/drivers/dma/mediatek/Kconfig
+@@ -36,3 +36,9 @@ config MTK_UART_APDMA
+ 	  When SERIAL_8250_MT6577 is enabled, and if you want to use DMA,
+ 	  you can enable the config. The DMA engine can only be used
+ 	  with MediaTek SoCs.
++
++config MTK_HSDMA
```

Comment:
is the driver still used? 
- The key name `MTK_HSDMA` is already used by the upstream MediaTek HSDMA driver which lacks MT7621 support. 
- From the mt7621.dtsi the driver is disabled. [2]
- In the final kernel config the DMA subsystem is disabled. [3]


1.  The `drivers/dma/mediatek/Kconfig`
```
# SPDX-License-Identifier: GPL-2.0-only

config MTK_HSDMA                                                                                                                     
▸       tristate "MediaTek High-Speed DMA controller support"
▸       depends on ARCH_MEDIATEK || COMPILE_TEST
▸       select DMA_ENGINE
▸       select DMA_VIRTUAL_CHANNELS
▸       help
▸         Enable support for High-Speed DMA controller on MediaTek
▸         SoCs.

▸         This controller provides the channels which is dedicated to
▸         memory-to-memory transfer to offload from CPU through ring-
▸         based descriptor management.

config MTK_CQDMA
▸       tristate "MediaTek Command-Queue DMA controller support"
▸       depends on ARCH_MEDIATEK || COMPILE_TEST
▸       select DMA_ENGINE
▸       select DMA_VIRTUAL_CHANNELS
▸       select ASYNC_TX_ENABLE_CHANNEL_SWITCH
▸       help
▸         Enable support for Command-Queue DMA controller on MediaTek
▸         SoCs.

▸         This controller provides the channels which is dedicated to
▸         memory-to-memory transfer to offload from CPU.

config MTK_UART_APDMA
▸       tristate "MediaTek SoCs APDMA support for UART"
▸       depends on OF && SERIAL_8250_MT6577
▸       select DMA_ENGINE
▸       select DMA_VIRTUAL_CHANNELS
▸       help
▸         Support for the UART DMA engine found on MediaTek MTK SoCs.
▸         When SERIAL_8250_MT6577 is enabled, and if you want to use DMA,
▸         you can enable the config. The DMA engine can only be used
▸         with MediaTek SoCs.

config MTK_HSDMA
▸       tristate "MTK HSDMA support"
▸       depends on RALINK && SOC_MT7621
▸       select DMA_ENGINE
▸       select DMA_VIRTUAL_CHANNELS
~                                             
```
2. The `DMA_ENGINE` and `DMA_VIRTUAL_CHANNELS` are both disabled in kconf.
```
$ grep "DMA_ENGINE" build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-6.1.38/.config

```
3. MT7621 dtsi file:
```

▸       ▸       hsdma: hsdma@7000 {
▸       ▸       ▸       compatible = "mediatek,mt7621-hsdma";
▸       ▸       ▸       reg = <0x7000 0x1000>;

▸       ▸       ▸       resets = <&sysc MT7621_RST_HSDMA>;
▸       ▸       ▸       reset-names = "hsdma";

▸       ▸       ▸       interrupt-parent = <&gic>;
▸       ▸       ▸       interrupts = <0 11 IRQ_TYPE_LEVEL_HIGH>;

▸       ▸       ▸       #dma-cells = <1>;
▸       ▸       ▸       #dma-channels = <1>;
▸       ▸       ▸       #dma-requests = <1>;

▸       ▸       ▸       status = "disabled";
▸       ▸       };
▸       };
```

Reply:
Yes I think you are right.